htdocs/projet/activity/perweek.php

```diff
@@ -307,7 +307,7 @@
 				}
 			}
 
-			if (!$updateoftaskdone) {  // Check to update progress if no update were done on task.
+			if (!$updateoftaskdone && GETPOSTISSET($tmptaskid.'progress')) {  // Check to update progress if no update were done on task.
```

Comment:
The fix was for 20.0, but in 17.0 the variable name is different.
```suggestion
			if (!$updateoftaskdone && GETPOSTISSET($taskid.'progress')) {  // Check to update progress if no update were done on task.
```